neo/core/filters.py

```diff
@@ -0,0 +1,109 @@
+"""
+This module implements :class:`FilterCondition`, which enables use of different filter conditions for
+neo.core.container.filter.
+"""
+
+
+class FilterCondition:
+    """
+        FilterCondition object is given as parameter to container.filter():
+
+        segment.filter(my_annotation=<FilterCondition>) or
+        segment=filter({'my_annotation': <FilterCondition>})
+    """
+
+    def __init__(self, z):
+        pass
+
+    def evaluate(self, x):
+        raise NotImplementedError()
+
+
+class Equal(FilterCondition):
```

Comment:
I would call this `Equals` or `EqualTo`

Reply:
See: 21f829af55ae3861942c58c2e6419152227476c1